src/components/DiagramContainer.tsx

```diff
@@ -93,28 +64,174 @@ class DiagramContainer extends React.Component<DiagramProps, DiagramState> {
+          console.log("node was inserted");
+          console.log(nodeData);
+
+          console.log("Sending POST request");
+          // TODO: use loc instead of x, y
+          fetch(
+            "https://localhost:7009/graph/vertices?" +
+              new URLSearchParams({
+                key: nodeData["key"],
+                type: nodeData["category"],
+                x: "1",
+                y: "5",
+              }),
+            { method: "POST" }
+          )
+            .then((response) => {
+              if (response.ok) {
+                console.log(`Successfully added node on backend`, nodeData);
+              } else {
+                throw new Error(
+                  JSON.stringify({
+                    status: response.status,
+                    body: response.text(),
+                  })
+                );
+              }
+            })
+            .catch((err) => console.log(err));
+        }
+        // node modified
+        else {
+          // TODO: update position with PUT request
+        }
+      }
+    }
+
+    // node removed
+    if (removedNodeKeys) {
+      for (let removedNodeKey of removedNodeKeys) {
+        if (removedNodeKey === undefined) {
+          continue;
+        }
+
+        fetch(
+          "https://localhost:7009/graph/vertices?" +
+            new URLSearchParams({
+              key: removedNodeKey.toString(),
+            }),
+          { method: "DELETE" }
+        )
+          .then((response) => {
+            if (response.ok) {
+              console.log(
+                `Successfully deleted key ${removedNodeKey} on backend`
+              );
+            } else {
+              throw new Error(
+                JSON.stringify({
+                  status: response.status,
+                  body: response.text(),
+                })
+              );
+            }
+          })
+          .catch((err) => console.log(err));
+      }
+    }
+
+    if (modifiedLinkData) {
+      console.log(modifiedLinkData);
+
+      for (let modifiedLink of modifiedLinkData) {
+        if (
+          insertedLinkKeys &&
+          insertedLinkKeys.includes(modifiedLink["key"])
+        ) {
+          fetch(
+            "https://localhost:7009/graph/edges?" +
+              new URLSearchParams({
+                srcKey: modifiedLink["from"].toString(),
+                dstKey: modifiedLink["to"].toString(),
+              }),
+            { method: "POST" }
+          )
+            .then((response) => {
+              if (response.ok) {
+                console.log(`Successfully added edge on backend`);
+                console.log(modifiedLink);
+              } else {
+                throw new Error(
+                  JSON.stringify({
+                    status: response.status,
+                    body: response.text(),
+                  })
+                );
+              }
+            })
+            .catch((err) => console.log(err));
+        }
```

Comment:
I believe you're missing an else and a todo here, for when a link is modified like what you did on line 133.

Reply:
because links can be modified to change their srcKey and dstKey.